src/care/LoopFuser.cpp

```diff
@@ -59,20 +59,16 @@ LoopFuser * LoopFuser::getInstance() {
 LoopFuser::~LoopFuser() {
    warnIfNotFlushed();
    if (m_reserved > 0) {
-#ifdef __CUDACC__
-      cudaFree(m_action_offsets);
-#elif defined(__HIPCC__)
-      hipFree(m_action_offsets);
+#if defined(CARE_GPUCC)
+      care::gpuFree(m_action_offsets);
```

Comment:
I wonder if we should just write a function in the care namespace called free and have it take a memory space as an argument? This will require some thought, though. Maybe we'll end up using camp or something. Probably not something for this pull request, though.

Reply:
Hi Alan. That is actually a good idea. Googling a bit, we actually do NOT even have to pass the memory space as an argument, since the function 

cudaPointerGetAttributes (and similar function for HIP)

can query whether a pointer is allocated on the host or device or managed. I'll add that as a future task